controllers/numaresourcesoperator_controller.go

```diff
@@ -143,19 +143,22 @@ func (r *NUMAResourcesOperatorReconciler) Reconcile(ctx context.Context, req ctr
 		return r.updateStatus(ctx, instance, status.ConditionDegraded, status.ConditionTypeIncorrectNUMAResourcesOperatorResourceName, err)
 	}
 
-	if err := validation.NodeGroups(instance.Spec.NodeGroups); err != nil {
+	if err := validation.NodeGroups(instance.Spec.NodeGroups, r.Platform); err != nil {
 		return r.updateStatus(ctx, instance, status.ConditionDegraded, validation.NodeGroupsError, err)
 	}
 
-	trees, err := getTreesByNodeGroup(ctx, r.Client, instance.Spec.NodeGroups)
+	trees, err := getTreesByNodeGroup(ctx, r.Client, instance.Spec.NodeGroups, r.Platform)
 	if err != nil {
 		return r.updateStatus(ctx, instance, status.ConditionDegraded, validation.NodeGroupsError, err)
 	}
 
-	multiMCPsErr := validation.MultipleMCPsPerTree(instance.Annotations, trees)
+	var multiMCPsErr error
+	if r.Platform == platform.OpenShift {
+		multiMCPsErr = validation.MultipleMCPsPerTree(instance.Annotations, trees)
 
-	if err := validation.MachineConfigPoolDuplicates(trees); err != nil {
-		return r.updateStatus(ctx, instance, status.ConditionDegraded, validation.NodeGroupsError, err)
+		if err := validation.MachineConfigPoolDuplicates(trees); err != nil {
+			return r.updateStatus(ctx, instance, status.ConditionDegraded, validation.NodeGroupsError, err)
+		}
```

Comment:
all the validation code should be abstracted (in the aforementioned future cleanup once CI runs) in a platform-specific function instead of having plenty of `if platform == ` like we have today